torch_geometric/nn/models/neural_fingerprint.py

```diff
@@ -0,0 +1,73 @@
+import torch
+import torch.nn as nn
+
+from torch_geometric.nn import MessagePassing
+from torch_geometric.utils import add_self_loops
+
+
+class NeuralConv(MessagePassing):
+    def __init__(self, num_features, fingerprint_length):
+        super(NeuralConv, self).__init__(aggr="add")
+        self.H = nn.ModuleList()
+        # H[i] would correspond to the hidden weight which would be used
+        # when num_neighbors=i+1
+        for i in range(5):
```

Comment:
The max degree(5) shouldn't be hard-coded in. You should pass it in as a parameter.

Reply:
You are right, but in the paper when they were implementing the model they hardcoded it as 5, because max neighbours of an atom in an organic molecule can go up to 5, so just for the sake of following the paper I hardcoded it as 5.

Reply:
Still think you should change it, in case a molecular graph is being passed in that has a max degree higher than 5.